cardea/modeling/modeler.py

```diff
@@ -1,506 +1,284 @@
 
-        self.primitive = self.check_path(primitives)
+        if hyperparameters:
+            model_instance.set_hyperparameters(hyperparameters)
 
-        if hyperparameters is not None:
-            hyperparameters = self.check_path_hyperparameters(hyperparameters)
-            pipeline = MLPipeline(self.primitive, hyperparameters)
+        if self._problem_type == 'regression':
+            scorer = self.regression_metrics[scoring or 'R2 Score']
         else:
-            pipeline = MLPipeline(self.primitive)
-        return pipeline
+            scorer = self.classification_metrics[scoring or 'F1 Macro']
 
-    def fit_predict_model(self, X_train, y_train, X_test, pipeline):
-        """Fits and predicts all the primitives within the pipeline.
+        scores = []
+        kf = KFold(n_splits=10, random_state=None, shuffle=True)
+        for train_index, test_index in kf.split(X):
+            model_instance.fit(X.iloc[train_index], y.iloc[train_index])
+            y_pred = model_instance.predict(X.iloc[test_index])
+            scores.append(scorer(y.iloc[test_index], y_pred))
 
-        Args:
-            X_train: A ndarray of the training data.
-            y_train: An array of the training target variable.
-            X_test: A ndarray of the testing data.
-            pipeline: A MLPipeline instance.
+        return np.mean(scores)
 
-        Returns:
-            A list consists of the used pipeline and an array of the predicted values.
-        """
-        pipeline.fit(X_train, y_train)
-        y_pred = pipeline.predict(X_test)
-        return y_pred
-
-    def search_all_possible_primitives(
-            self, primitives, hyperparameters=None):
-        """Searches for all primitives similar to the ones provided.
+    def tune(self, X, y, max_evals=10, scoring=None, verbose=False):
+        """ Tune the pipeline hyper-parameters and select the optimized model.
 
         Args:
-            primitives: A list of primitive.
-            hyperparameters: A dictionary of hyperparameters for each primitives.
-
-        Returns:
-            A list that encapsulate a list consisting of the fold number and the used pipeline and
-            an array of the predicted values and an array of the actual values.
+            X (array-like):
+                Inputs to the pipeline.
+            y (array-like):
+                Target values.
+            max_evals (int):
+                Maximum number of hyper-parameter optimization iterations.
+            scoring (str):
+                The name of the scoring function.
+            verbose (bool):
+                Whether to log information during processing.
         """
+        tunables = {'0': self._pipeline.get_tunable_hyperparameters(flat=True)}
 
-        pipeline_list = []
-        pipeline_list.append(
-            self.create_kfold(primitives))
+        session = BTBSession(tunables, lambda _, hyparam: self.k_fold_validation(
```

Comment:
The `scorer(tunable_name, config)` function as an input to the BTBSessions takes the `tunable_name` as its first parameter. This parameter is used to specify which pipeline is scoring. We don't need it because we only have one pipeline to tune and select from.